lib/ansible/config/base.yml

```diff
@@ -589,7 +589,7 @@ DEFAULT_GATHERING:
   choices: ['smart', 'explicit', 'implicit']
 DEFAULT_GATHER_SUBSET:
   name: Gather facts subset
-  default: 'all'
+  default: ['all']
```

Comment:
How does this interact with https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/play_context.py#L188 ?  That seems to assume DEFAULT_GATHER_SUBSET is a string. (Though, everywhere in module_utils.facts seems to assumes gather_subset is or should be a list, so changing the default sounds right)